test/unit/abilities/provider_any_test.rb

```diff
@@ -71,57 +81,112 @@ def test_policies_not_tenant
         assert_cannot ability, :show, cinstance_event
       end
     end
+  end
 
-    test 'AccountRelatedEvent can show if has :partners and does not have a service' do
-      assert_cannot ability, :show, Accounts::AccountCreatedEvent.create(account, user)
+  class AccountRelatedEventTest < BaseTest
+    setup do
+      # @account has a service and a service plan
+      @service = FactoryBot.create(:simple_service, account: @account)
+      service_plan = FactoryBot.create(:simple_service_plan, issuer: @service)
+
+      # There's a buyer for @account and it's subscribed to the service
+      @buyer = FactoryBot.create(:buyer_account, provider_account: @account)
+      @buyer_user = FactoryBot.create(:simple_user, account: @buyer)
+      FactoryBot.create(:service_contract, plan: service_plan, user_account: @buyer)
+
+      # Another buyer but not subscribed to any service
+      @buyer_no_service =  FactoryBot.create(:buyer_account, provider_account: @account)
+      @buyer_no_service_user = FactoryBot.create(:simple_user, account: @buyer_no_service)
+    end
 
+    test "member has :partners can't show AccountRelatedEvent when there is no service" do
+      user.member_permission_service_ids = nil # All services allowed
       user.member_permission_ids = [:partners]
-      assert_can ability, :show, Accounts::AccountCreatedEvent.create(account, user)
+      assert_cannot ability, :show, Accounts::AccountCreatedEvent.create(@buyer_no_service, @buyer_no_service_user)
```

Comment:
Hmm.. I understand why this happens, but not sure I agree with this behavior :thinking: 

Intuitively, I would expect the user that has permission for ALL services be able to see the AccountCreatedEvents, even if the related account is not signed up to any service. Because if they were signed up - to **any** service - the buyer would be able to see it.

Reply:
What you say makes sense, I'll take another look at this.

Reply:
What about this? https://github.com/3scale/porta/pull/4029/commits/afcab73c6b387b8c5eff372caaf6bad4ab2906b8